src/ore/src/lgbytes.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The [bytes] crate but backed by [lgalloc].
+
+use std::sync::Arc;
+
+use bytes::Buf;
+use lgalloc::AllocError;
+
+use crate::region::Region;
+
+/// [bytes::Bytes] but backed by [lgalloc].
+#[derive(Clone)]
+pub struct LgBytes {
+    offset: usize,
+    region: Arc<Region<u8>>,
```

Comment:
Once it lands, we probably want to switch to `Array` from https://github.com/MaterializeInc/materialize/pull/23944/files#diff-b34b0ca4eb3ec3790fcbc7dc4698ab0bd77a02d7d7fb04a83134aff667f4cc62R115. No need to block this PR, though.

Reply:
I hadn't seen that yet, what's the advantage?